src/components/PopupWithForm.js

```diff
@@ -28,6 +28,7 @@ export default class PopupWithForm extends Popup {
       evt.preventDefault();
       this._handleFormSubmit(this._getInputValues());
       this._popupForm.reset();
+      this._submitButton.textContent = "Saving...";
```

Comment:
Needs correction: please don't change the button text from here as the form can be submitted without an api request. Manage loading from `index.js`. It's important to start loading right before the api call and end loading in promise's `finally` block (https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/finally), because loading should end even if an error occurred

One of the options to implement that would be adding a parameter `loadingButtonText` to the constructor of `PopupWithForm` and using it to change button text. 

```js
constructor(popupSelector, { handleSubmit, loadingButtonText }) { 
  ...
  this._submitButton = this._formElement.querySelector(...)
  this._buttonText = this._submitButton.textContent;
  this._loadingButtonText = loadingButtonText;
  …
}

showLoading() {
  this._submitButton.textContent = this._loadingButtonText
}

hideLoading() {
  this._submitButton.textContent = this._buttonText
}
``` 

The api call should look like this:

```js
popup.showLoading()
api.
  .then(...)
  .catch(...)
  .finally(() => {
    popup.hideLoading()

  })
```

The same thing for the confirmation popup